colm/transpile/transpile.lm

```diff
@@ -0,0 +1,265 @@
+namespace flux
+	include '../flux.lm'
+end
+
+include '../influxql.lm'
+
+if argv->length < 1 {
+	send stderr "usage: ./transpile bucket-name
+	exit(1)
+}
+
+parse P: influxql [stdin] 
+if ( !P ) {
+	send stderr "error parsing influxql query: [error]
+	exit( 1 )
+}
+
+bool time_constraint( Any: any )
+{
+	for UE: unary_expr in Any {
+		if match UE "time"
+			return true
+	}
+	return false
+}
+
+def time_range
+	Start: flux::expression
+	Stop: flux::expression
+[]
+
+void unquote_times( AE: ref<additive_expression> )
+{
+	for QI: quoted_identifier in AE {
+		Time: str = $QI
+		Time = prefix( suffix( Time, 1 ), Time.length - 2 )
+		QI.data = Time
+	}
+}
+
+bool check_time( Range: ref<time_range>, CE: comparison_expression )
```

Comment:
What does `ref` mean?

Reply:
Colm passes all trees around as values. Under the hood it uses refcounting to do this efficiently. So if you pass a tree to a function and then iterate through the tree and modify it, then return, the modifications are lost.

The ref<type> syntax gives you a reference. A mutable reference, to be exact :)